### PR TITLE
Honour the WEBHOOK_TLS_CERT_DIR env. variable for finding the Webhook TLS certificate

### LINKED ISSUE #1147: [Bug]: Can't install provider-aws 1.0.0 on crossplane versions before 1.14
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Affected Resource(s)

_No response_

### Resource MRs required to reproduce the bug

_No response_

### Steps to Reproduce

Create a kind cluster. 
Install uxp [v1.13.2-up.3](https://github.com/upbound/universal-crossplane/releases/tag/v1.13.2-up.3)
Install provider-aws v0.47.1, including at least one of the provider packages that have conversion webhooks (ec2, rds, elasticache, or a couple others)
Wait for the providers to stabilize
Upgrade the providers to provider-aws  v1.0.0

### What happened?

The providers which have conversion webhooks got into a crashloop, and never became ready. ec2, rds, elasticache, etc resources become completely unusable. 

### Relevant Error Output Snippet

```shell
From provider-aws-elasticache v1.0.0:
2024-02-08T02:22:36Z	DEBUG	provider-aws	Starting	{"sync-interval": "1h0m0s", "poll-interval": "10m0s", "poll-jitter": "30s", "max-reconcile-rate": 100}
2024-02-08T02:22:38Z	INFO	provider-aws	Beta feature enabled	{"flag": "EnableBetaManagementPolicies"}
2024-02-08T02:22:38Z	INFO	controller-runtime.builder	skip registering a mutating webhook, object does not implement admission.Defaulter or WithDefaulter wasn't called	{"GVK": "elasticache.aws.upbound.io/v1beta1, Kind=Cluster"}
2024-02-08T02:22:38Z	INFO	controller-runtime.builder	skip registering a validating webhook, object does not implement admission.Validator or WithValidator wasn't called	{"GVK": "elasticache.aws.upbound.io/v1beta1, Kind=Cluster"}
2024-02-08T02:22:38Z	INFO	controller-runtime.builder	skip registering a mutating webhook, object does not implement admission.Defaulter or WithDefaulter wasn't called	{"GVK": "elasticache.aws.upbound.io/v1beta1, Kind=ParameterGroup"}
2024-02-08T02:22:38Z	INFO	controller-runtime.builder	skip registering a validating webhook, object does not implement admission.Validator or WithValidator wasn't called	{"GVK": "elasticache.aws.upbound.io/v1beta1, Kind=ParameterGroup"}
2024-02-08T02:22:38Z	INFO	controller-runtime.builder	skip registering a mutating webhook, object does not implement admission.Defaulter or WithDefaulter wasn't called	{"GVK": "elasticache.aws.upbound.io/v1beta2, Kind=ReplicationGroup"}
2024-02-08T02:22:38Z	INFO	controller-runtime.builder	skip registering a validating webhook, object does not implement admission.Validator or WithValidator wasn't called	{"GVK": "elasticache.aws.upbound.io/v1beta2, Kind=ReplicationGroup"}
2024-02-08T02:22:38Z	INFO	controller-runtime.webhook	Registering webhook	{"path": "/convert"}
2024-02-08T02:22:38Z	INFO	controller-runtime.builder	Conversion webhook enabled	{"GVK": "elasticache.aws.upbound.io/v1beta2, Kind=ReplicationGroup"}
2024-02-08T02:22:38Z	INFO	controller-runtime.builder	skip registering a mutating webhook, object does not implement admission.Defaulter or WithDefaulter wasn't called	{"GVK": "elasticache.aws.upbound.io/v1beta1, Kind=SubnetGroup"}
2024-02-08T02:22:38Z	INFO	controller-runtime.builder	skip registering a validating webhook, object does not implement admission.Validator or WithValidator wasn't called	{"GVK": "elasticache.aws.upbound.io/v1beta1, Kind=SubnetGroup"}
2024-02-08T02:22:38Z	INFO	controller-runtime.builder	skip registering a mutating webhook, object does not implement admission.Defaulter or WithDefaulter wasn't called	{"GVK": "elasticache.aws.upbound.io/v1beta1, Kind=User"}
2024-02-08T02:22:38Z	INFO	controller-runtime.builder	skip registering a validating webhook, object does not implement admission.Validator or WithValidator wasn't called	{"GVK": "elasticache.aws.upbound.io/v1beta1, Kind=User"}
2024-02-08T02:22:38Z	INFO	controller-runtime.builder	skip registering a mutating webhook, object does not implement admission.Defaulter or WithDefaulter wasn't called	{"GVK": "elasticache.aws.upbound.io/v1beta1, Kind=UserGroup"}
2024-02-08T02:22:38Z	INFO	controller-runtime.builder	skip registering a validating webhook, object does not implement admission.Validator or WithValidator wasn't called	{"GVK": "elasticache.aws.upbound.io/v1beta1, Kind=UserGroup"}
2024-02-08T02:22:38Z	INFO	controller-runtime.metrics	Starting metrics server
2024-02-08T02:22:38Z	INFO	controller-runtime.webhook	Starting webhook server
2024-02-08T02:22:38Z	INFO	Stopping and waiting for non leader election runnables
2024-02-08T02:22:38Z	INFO	Stopping and waiting for leader election runnables
2024-02-08T02:22:38Z	INFO	controller-runtime.metrics	Serving metrics server	{"bindAddress": ":8080", "secure": false}
2024-02-08T02:22:38Z	INFO	Starting EventSource	{"controller": "managed/elasticache.aws.upbound.io/v1beta1, kind=cluster", "source": "kind source: *v1beta1.Cluster"}
2024-02-08T02:22:38Z	INFO	Starting Controller	{"controller": "managed/elasticache.aws.upbound.io/v1beta1, kind=cluster"}
2024-02-08T02:22:38Z	INFO	Starting EventSource	{"controller": "managed/elasticache.aws.upbound.io/v1beta2, kind=replicationgroup", "source": "kind source: *v1beta2.ReplicationGroup"}
2024-02-08T02:22:38Z	INFO	Starting Controller	{"controller": "managed/elasticache.aws.upbound.io/v1beta2, kind=replicationgroup"}
2024-02-08T02:22:38Z	INFO	Starting workers	{"controller": "managed/elasticache.aws.upbound.io/v1beta2, kind=replicationgroup", "worker count": 100}
2024-02-08T02:22:38Z	INFO	Starting EventSource	{"controller": "managed/elasticache.aws.upbound.io/v1beta1, kind=usergroup", "source": "kind source: *v1beta1.UserGroup"}
2024-02-08T02:22:38Z	INFO	Starting Controller	{"controller": "managed/elasticache.aws.upbound.io/v1beta1, kind=usergroup"}
2024-02-08T02:22:38Z	INFO	Starting workers	{"controller": "managed/elasticache.aws.upbound.io/v1beta1, kind=usergroup", "worker count": 100}
2024-02-08T02:22:38Z	INFO	Shutdown signal received, waiting for all workers to finish	{"controller": "managed/elasticache.aws.upbound.io/v1beta2, kind=replicationgroup"}
2024-02-08T02:22:38Z	INFO	Starting EventSource	{"controller": "managed/elasticache.aws.upbound.io/v1beta1, kind=parametergroup", "source": "kind source: *v1beta1.ParameterGroup"}
2024-02-08T02:22:38Z	INFO	Starting Controller	{"controller": "managed/elasticache.aws.upbound.io/v1beta1, kind=parametergroup"}
2024-02-08T02:22:38Z	INFO	Starting workers	{"controller": "managed/elasticache.aws.upbound.io/v1beta1, kind=parametergroup", "worker count": 100}
2024-02-08T02:22:38Z	INFO	Starting workers	{"controller": "managed/elasticache.aws.upbound.io/v1beta1, kind=cluster", "worker count": 100}
2024-02-08T02:22:38Z	INFO	Starting EventSource	{"controller": "managed/elasticache.aws.upbound.io/v1beta1, kind=user", "source": "kind source: *v1beta1.User"}
2024-02-08T02:22:38Z	INFO	Starting Controller	{"controller": "managed/elasticache.aws.upbound.io/v1beta1, kind=user"}
2024-02-08T02:22:38Z	INFO	Starting workers	{"controller": "managed/elasticache.aws.upbound.io/v1beta1, kind=user", "worker count": 100}
2024-02-08T02:22:38Z	INFO	Shutdown signal received, waiting for all workers to finish	{"controller": "managed/elasticache.aws.upbound.io/v1beta1, kind=usergroup"}
2024-02-08T02:22:38Z	INFO	Shutdown signal received, waiting for all workers to finish	{"controller": "managed/elasticache.aws.upbound.io/v1beta1, kind=parametergroup"}
2024-02-08T02:22:38Z	INFO	All workers finished	{"controller": "managed/elasticache.aws.upbound.io/v1beta2, kind=replicationgroup"}
2024-02-08T02:22:38Z	INFO	Shutdown signal received, waiting for all workers to finish	{"controller": "managed/elasticache.aws.upbound.io/v1beta1, kind=cluster"}
2024-02-08T02:22:38Z	INFO	Shutdown signal received, waiting for all workers to finish	{"controller": "managed/elasticache.aws.upbound.io/v1beta1, kind=user"}
2024-02-08T02:22:38Z	INFO	All workers finished	{"controller": "managed/elasticache.aws.upbound.io/v1beta1, kind=parametergroup"}
2024-02-08T02:22:38Z	INFO	All workers finished	{"controller": "managed/elasticache.aws.upbound.io/v1beta1, kind=cluster"}
2024-02-08T02:22:38Z	INFO	Starting EventSource	{"controller": "managed/elasticache.aws.upbound.io/v1beta1, kind=subnetgroup", "source": "kind source: *v1beta1.SubnetGroup"}
2024-02-08T02:22:38Z	INFO	Starting Controller	{"controller": "managed/elasticache.aws.upbound.io/v1beta1, kind=subnetgroup"}
2024-02-08T02:22:38Z	INFO	All workers finished	{"controller": "managed/elasticache.aws.upbound.io/v1beta1, kind=usergroup"}
2024-02-08T02:22:38Z	INFO	Starting workers	{"controller": "managed/elasticache.aws.upbound.io/v1beta1, kind=subnetgroup", "worker count": 100}
2024-02-08T02:22:38Z	INFO	Shutdown signal received, waiting for all workers to finish	{"controller": "managed/elasticache.aws.upbound.io/v1beta1, kind=subnetgroup"}
2024-02-08T02:22:38Z	INFO	All workers finished	{"controller": "managed/elasticache.aws.upbound.io/v1beta1, kind=user"}
2024-02-08T02:22:38Z	INFO	All workers finished	{"controller": "managed/elasticache.aws.upbound.io/v1beta1, kind=subnetgroup"}
2024-02-08T02:22:38Z	INFO	Stopping and waiting for caches
2024-02-08T02:22:38Z	ERROR	controller-runtime.source.EventHandler	failed to get informer from cache	{"error": "Timeout: failed waiting for *v1beta1.Cluster Informer to sync"}
sigs.k8s.io/controller-runtime/pkg/internal/source.(*Kind).Start.func1.1
	sigs.k8s.io/controller-runtime@v0.16.3/pkg/internal/source/kind.go:68
k8s.io/apimachinery/pkg/util/wait.loopConditionUntilContext.func1
	k8s.io/apimachinery@v0.29.0/pkg/util/wait/loop.go:53
k8s.io/apimachinery/pkg/util/wait.loopConditionUntilContext
	k8s.io/apimachinery@v0.29.0/pkg/util/wait/loop.go:54
k8s.io/apimachinery/pkg/util/wait.PollUntilContextCancel
	k8s.io/apimachinery@v0.29.0/pkg/util/wait/poll.go:33
sigs.k8s.io/controller-runtime/pkg/internal/source.(*Kind).Start.func1
	sigs.k8s.io/controller-runtime@v0.16.3/pkg/internal/source/kind.go:56
2024-02-08T02:22:38Z	INFO	Stopping and waiting for webhooks
2024-02-08T02:22:38Z	INFO	Stopping and waiting for HTTP servers
2024-02-08T02:22:38Z	INFO	controller-runtime.metrics	Shutting down metrics server with timeout of 1 minute
2024-02-08T02:22:38Z	INFO	Wait completed, proceeding to shutdown the manager
provider: error: Cannot start controller manager: open /tls/server/tls.crt: no such file or directory
```

### Crossplane Version

v1.13.2-up.3 and earlier (I also tested v1.13.2-up.1 and v1.12.3-up.1)

### Provider Version

0.47.1 upgrading to 1.0.0

### Kubernetes Version

_No response_

### Kubernetes Distribution

Both EKS and Kind

### Additional Info

_No response_

### PR DESCRIPTION
**Pull request by ulucinar**
### Description of your changes

Fixes #1147 

Crosslane versions before v1.14 [mount the Webhook TLS certificate and key](https://github.com/upbound/provider-aws/issues/1147#issuecomment-1938350942) at a path other than the current default `/tls/certs` and make that location available via the environment variable `WEBHOOK_TLS_CERT_DIR`. `provider-aws` currently relies on the default path and the `CERTS_DIR` environment variable, breaking installations prior to `v1.14`. This PR implements the following protocol for configuring the conversion Webhook TLS certificate & key configuration:
1. If the `--certs-dir` command-line option is supplied, it's used.
1. If the `--certs-dir` command-line option is *not* supplied, the following environment variables are used in the given order: `CERTS_DIR` (for backwards-compatibility reasons), `TLS_SERVER_CERTS_DIR` (the new environment variable, which has replaced the `WEBHOOK_TLS_CERT_DIR` env. variable in Crossplane), and `WEBHOOK_TLS_CERT_DIR` (for backwards-compatibility).

This PR also fixes the example manifest `examples/ec2/v1beta2/route.yaml`, where a non-existent version of `RouteTable.ec2` is referenced.

I have:

- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

Successfully provisioned a `Route.ec2/v1beta2` resource using `examples/ec2/v1beta2/route.yaml` with the following setup:
- UXP version: `v1.13.2-up.3`
- Provider package: `index.docker.io/ulucinar/provider-aws-ec2:v0.48.0-ecf7ae9ad372b96dad94e52f6e7aae9b6db44063` (built from this PR's feature branch)

Also validated using:
- UXP version: `v1.14.5-up.1`
- Provider package: `index.docker.io/ulucinar/provider-aws-ec2:v0.48.0-ecf7ae9ad372b96dad94e52f6e7aae9b6db44063`

Uptest run for the modified example manifest `examples/ec2/v1beta2/route.yaml`: https://github.com/upbound/provider-aws/actions/runs/7917023305/

### CODE REVIEW

**Comment by ulucinar** [comment]
/test-examples="examples/ec2/v1beta1/vpc.yaml"

File: hack/main.go.tmpl [hunk 1]
```diff
@@ -60,7 +67,11 @@ func main() {
 		essTLSCertsPath            = app.Flag("ess-tls-cert-dir", "Path of ESS TLS certificates.").Envar("ESS_TLS_CERTS_DIR").String()
 		enableManagementPolicies   = app.Flag("enable-management-policies", "Enable support for Management Policies.").Default("true").Envar("ENABLE_MANAGEMENT_POLICIES").Bool()
 
-		certsDir = app.Flag("certs-dir", "The directory that contains the server key and certificate.").Default("/tls/server").Envar("CERTS_DIR").String()
+		certsDirSet = false
+		certsDir    = app.Flag("certs-dir", "The directory that contains the server key and certificate.").Default(tlsServerCertDir).Envar(certsDirEnvVar).PreAction(func(_ *kingpin.ParseContext) error {
```
Comment (sergenyalcin) [inline hack/main.go.tmpl:71]: nit: As discussed offline, here we rely on calling the PreAction mechanism when the flag is set. I am writing this to record it for documentation purposes for the other contributors. We may consider putting a code comment here.

**Review by sergenyalcin — APPROVED** [review]
Thanks @ulucinar LGTM! Left a nit comment for documentation.

**Comment by ulucinar** [comment]
/test-examples="examples/ec2/v1beta2/route.yaml"

File: hack/main.go.tmpl (hunk 1 above)
Comment (ulucinar) [reply hack/main.go.tmpl]: Thanks @sergenyalcin. Added comments that explain the purpose of using a `PreAction` there.

**Comment by ulucinar** [comment]
/test-examples="examples/ec2/v1beta2/route.yaml"

**Pull request merged**